src/test/java/com/google/sps/PersistentUserRepositoryTest.java

```diff
@@ -79,17 +77,17 @@ public void tearDown() throws Exception {
   public void addFakeMentorsTest() {
     PersistentUserRepository testDataRepo = new PersistentUserRepository();
     addFakeMentors(testDataRepo);
-    String expected = "Andre Jerry Julie ";
-    String result = testDataRepo.toString();
+    int expected = 3;
+    int result = testDataRepo.getAllUsers().size();
     Assert.assertEquals(expected, result);
   }
 
   @Test
   public void addUserTest() {
     PersistentUserRepository emptyRepo = new PersistentUserRepository();
     emptyRepo.addUser(USER_A);
-    String expected = "User A ";
-    String result = emptyRepo.toString();
+    int expected = 1;
```

Comment:
I agree that using toString in the test (as was the case before) is unwieldy and unmaintainable now that it's returning a large and ever changing json object, rather than a string.

The wrapper tests, however, should test if each field in user is getting written and read back from datastore correctly (I've seen this exact error -- where someone forgets to serialize or deserialize a field -- pop up numerous times). Could you add checks for each of the fields (name, specialty, company, etc), to see if they're correctly read/written back?

This only really needs to be tested once for one user for us to be reasonably confident that serialization/deserialization isn't broken, so feel free to add it in this test or in another/new one.

Reply:
Done